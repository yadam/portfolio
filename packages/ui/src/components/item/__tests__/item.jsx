import axios from 'axios';
import React from 'react';
import renderer from 'react-test-renderer';
import ItemContainer from '..';

jest.mock('axios');
jest.mock('../component', () => 'Item');

let mockProps;

describe('Item Container', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    axios.get = jest.fn(
      () =>
        new Promise(resolve => {
          resolve({
            alt: 'Test alt',
            num: 'TestId',
            img: 'TestImageUrl',
            safe_title: 'TestTitle',
          });
        }),
    );
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ItemContainer {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('gets a comic', async () => {
    const tree = renderer.create(<ItemContainer {...mockProps} />);
    await tree.root.instance.componentDidMount();
    expect(tree).toMatchSnapshot();
  });
});
