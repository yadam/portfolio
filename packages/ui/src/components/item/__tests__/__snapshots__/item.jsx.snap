// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Container gets a comic 1`] = `
<Item
  alt="Test alt"
  id={123456}
  image="TestImageUrl"
  loading={false}
  onChange={[Function]}
  reaction=""
  title="TestTitle"
/>
`;

exports[`Item Container renders correctly 1`] = `
<Item
  alt=""
  id={0}
  image=""
  loading={true}
  onChange={[Function]}
  reaction=""
  title=""
/>
`;

exports[`Item Container sends a reaction 1`] = `
<Item
  alt="Test alt"
  id={123456}
  image="TestImageUrl"
  loading={false}
  onChange={[Function]}
  reaction="funny"
  title="TestTitle"
/>
`;
